lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchRuntimeConfigurationCreator.java

```diff
@@ -47,29 +44,43 @@
    * Get or create a {@link ConfigurationModule} for the Azure Batch runtime.
    *
    * @param isWindows true if Azure Batch pool nodes run Windows, false otherwise.
+   * @param isDockerContainer true if Azure Batch pool nodes are container-based, false otherwise.
    * @return the configuration module object.
    */
-  public static ConfigurationModule getOrCreateAzureBatchRuntimeConfiguration(final boolean isWindows) {
+  public static ConfigurationModule getOrCreateAzureBatchRuntimeConfiguration(
+      final boolean isWindows,
+      final boolean isDockerContainer) {
 
     if (AzureBatchRuntimeConfigurationCreator.conf == null) {
       ConfigurationModuleBuilder builder = AzureBatchRuntimeConfigurationStatic.CONF;
-      ConfigurationModule module;
+
       if (isWindows) {
-        module = builder.bindImplementation(CommandBuilder.class, WindowsCommandBuilder.class).build();
+        builder = builder.bindImplementation(CommandBuilder.class, WindowsCommandBuilder.class);
       } else {
-        module = builder.bindImplementation(CommandBuilder.class, LinuxCommandBuilder.class).build();
+        builder = builder.bindImplementation(CommandBuilder.class, LinuxCommandBuilder.class);
+      }
+
+      if (isDockerContainer) {
+        builder = builder.bindImplementation(TcpPortProvider.class, ListTcpPortProvider.class);
```

Comment:
This should not be necessary. The binding will be done by the user when they specify the TcpPorts.

Reply:
Got rid of it. This is along the same lines that we discussed yesterday where in the client was using the port provider that was meant for the driver/evaluator. In moving out the port provider to a separate place (HelloReefAzBatch class), I was able to get rid of it here.

---
In reply to: [199968798](https://github.com/sharathmalladi/reef/pull/6#discussion_r199968798) [](ancestors = 199968798)